File: src/main/java/net/hypixel/api/request/Request.java
package net.hypixel.api.request;

import com.palechip.hudpixelmod.extended.util.LoggerHelper;
import net.hypixel.api.HypixelAPI;

import java.util.Map;

public class Request {

    private static final String BASE_URL = "https://api.hypixel.net/";

    private final RequestType requestType;
    private final Map<RequestParam, Object> params;

    public Request(RequestType requestType, Map<RequestParam, Object> params) {
        this.requestType = requestType;
        this.params = params;
    }

    public RequestType getRequestType() {
        return requestType;
    }

    public Map<RequestParam, Object> getParams() {
        return params;
    }

    public String getURL(HypixelAPI hypixelAPI) {
        String url = BASE_URL;

        url += requestType.getKey();
        url += "?" + RequestParam.KEY.getRequestVar().getValue() + "=" + hypixelAPI.getApiKey();

        for (Map.Entry<RequestParam, Object> entry : params.entrySet()) {
            url += "&" + entry.getKey().getRequestVar().getValue() + "=" + entry.getKey().serialize(entry.getValue());
        }

        LoggerHelper.logInfo("[APIrequest]: " + url);

        return url;
    }
}

File: src/main/java/net/hypixel/api/request/Request.kt
package net.hypixel.api.request

import net.hypixel.api.HypixelAPI

class Request(val requestType: RequestType?, val params: Map<RequestParam, Any>) {

    fun getURL(hypixelAPI: HypixelAPI): String {
        var url = BASE_URL

        url += requestType?.key
        url += "?" + RequestParam.KEY.queryField + "=" + hypixelAPI.apiKey0

        for (entry in params.entries) {
            url += "&" + entry.key.queryField + "=" + entry.key.serialize(entry.value)
        }

        return url
    }

    companion object {

        private val BASE_URL = "https://api.hypixel.net/"
    }
}
